rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp

```diff
@@ -340,9 +340,21 @@ error_code cellMsgDialogOpen2(u32 type, vm::cptr<char> msgString, vm::ptr<CellMs
 
 error_code cellMsgDialogOpen(u32 type, vm::cptr<char> msgString, vm::ptr<CellMsgDialogCallback> callback, vm::ptr<void> userData, vm::ptr<void> extParam)
 {
-	// Note: This function needs proper implementation, solve first argument "type" conflict with MsgDialogOpen2 in cellMsgDialog.h.
-	cellSysutil.todo("cellMsgDialogOpen(type=0x%x, msgString=%s, callback=*0x%x, userData=*0x%x, extParam=*0x%x)", type, msgString, callback, userData, extParam);
-	return cellMsgDialogOpen2(type, msgString, callback, userData, extParam);
+	cellSysutil.warning("cellMsgDialogOpen(type=0x%x, msgString=%s, callback=*0x%x, userData=*0x%x, extParam=*0x%x)", type, msgString, callback, userData, extParam);
+
+	// PS3 testing indicated that cellMsgDialogOpen does have quite a few differences in regards to the YesNo selection, plus is only a subset of functionality
+	// The easiest way is to just bittest these, and set the corresponding bits for the cellMsgDialogOpen2 function
+	u32 msg_dialog_open2_type = 0;
+	if (type & CELL_MSGDIALOG_DIALOG_TYPE_NORMAL)
+		msg_dialog_open2_type |= CELL_MSGDIALOG_TYPE_SE_TYPE_NORMAL;
```

Comment:
this type of formatting is not really readable.
This function is small enough to add some newlines after each expression or scope